File: Frontend/src/service/userService.js
import axios from "../axios";

export const getUsers = () => axios.get("/user");
export const getUserById = (id) => axios.get(`/user/${id}`);
export const addUser = (user) => {
    axios.post("admin/user", user);
}
export const updateUser = (id, user) => axios.put(`/admin/user/${id}`, user);
export const deleteUser = (id) => axios.delete(`/admin/user/${id}`);

export const loginUser = (credentials) => axios.post("/user/login", credentials);
    

File: lib-man/src/main/java/com/example/lib_man/service/UserService.java
package com.example.lib_man.service;

import com.example.lib_man.model.Users;
import com.example.lib_man.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    UserRepository usersRepository;

    public List<Users> getAllUsers() {
        return usersRepository.findByRole("user");
    }

    public Users getUserById(int id) {
        return usersRepository.findById(id).orElse(null);
    }


    public Users addUser(Users user) {
        return usersRepository.save(user);
    }

    public Users updateUser(int userid, Users user) {
        Users existing = usersRepository.findById(userid).orElseThrow();
        existing.setUname(user.getUname());
        existing.setEmail(user.getEmail());
        existing.setPassword(user.getPassword());
        existing.setRole(user.getRole());
        return usersRepository.save(existing);
    }

    public void deleteUser(int id) {
        usersRepository.deleteById(id);
    }

    public Users login(String email, String password) {
        Users user = usersRepository.findByEmail(email);
        if (user != null && user.getPassword().equals(password)) {
            return user;
        }
        return null;
    }
}
